.github/workflows/image-publish.yml

```diff
@@ -64,10 +71,35 @@ jobs:
       - name: Docker Build and Push
         run: |
           SHORT_SHA1=$(git rev-parse --short=7 HEAD)
-          docker buildx build --platform linux/amd64 -f build/dockerfiles/assembly.Dockerfile --push -t quay.io/che-incubator/che-code:insiders -t quay.io/che-incubator/che-code:next -t quay.io/che-incubator/che-code:insiders-${SHORT_SHA1} .
+          if [[ ${{ env.BRANCH_NAME }} == main ]]; then
+            docker buildx build --platform linux/amd64 -f build/dockerfiles/assembly.Dockerfile --push -t quay.io/che-incubator/che-code:insiders -t quay.io/che-incubator/che-code:next -t quay.io/che-incubator/che-code:insiders-${SHORT_SHA1} .
+          elif [[ ${{ env.BRANCH_NAME }} =~ ^7\.[0-9]+\.[0-9]+$ ]]; then
+            docker buildx build --platform linux/amd64 -f build/dockerfiles/assembly.Dockerfile --push -t quay.io/che-incubator/che-code:${{env.BRANCH_NAME}}  . 
+          fi
+  release-finish:
+    name: release-finish
+    needs: [build, assembly]
+    runs-on: ubuntu-20.04
+    if: (success() || failure()) && github.ref != 'refs/heads/main'
+    steps:
+      - name: Create failure MM message
+        if: failure()
+        run: |
+          echo "{\"text\":\":no_entry_sign: Che Code ${{ env.BRANCH_NAME }} release has failed: https://github.com/che-incubator/che-code/actions/workflows/release.yml\"}" > mattermost.json
+      - name: Create success MM message
+        run: |
+          echo "{\"text\":\":white_check_mark: Che Code ${{ env.BRANCH_NAME }} has been released: https://quay.io/che-incubator/che-code:${{ env.BRANCH_NAME }}\"}" > mattermost.json
+      - name: Send MM message
+        if: success() || failure()
+        uses: mattermost/action-mattermost-notify@1.1.0
+        env:
+          MATTERMOST_WEBHOOK_URL: ${{ secrets.MATTERMOST_WEBHOOK_URL }}
+          MATTERMOST_CHANNEL: eclipse-che-releases
+          MATTERMOST_USERNAME: che-bot
   dev:
     name: dev
```

Comment:
might want to rename this to "build-dev" for clarity; also @azatsarynnyy does the dev image need to be built BEFORE the other images? Is it like with Che-theia there che-theia depends on che-theia-dev?

Reply:
no, Che-Code image doesn't depend on the dev image
so, the build order doesn't matter